lib/data/landing_page_content_items/landing_page.yaml

```diff
@@ -85,6 +85,8 @@ blocks:
     content: <p>Right content!</p>
 - type: govspeak
   content: <h2>Statistics</h2>
+  margin_top: 5
+  margin_bottom: 5
```

Comment:
I think we want to avoid having margins specified in the content item, because that implies that this is something that will be configured by the user, which I don't think we would want (could potentially break the page design).

Govspeak is a slightly tricky one, but maybe something that could be handled internally to the govspeak component. For the other blocks that need bottom margin, maybe the approach could be to remove their existing margins and rewrite your bottom margin class as a mixin, then just add it to their Sass?

Reply:
Yes I did weigh this up but felt that given possible uncertainly over final content, it might be useful for content editors to have some control of how much spacing to apply around govspeak elements. As you note, there's always the possibility of that control causing design issues. 

For your other suggestion, do you mean avoid the use of the named class `app-margin-bottom-responsive` and instead use a mixin? This seems less flexible to me, but perhaps it's just apples and oranges.

Reply:
To a degree, this gets us into the heart of the problem with this whole "blocks" approach.

If the user of the publishing system is allowed to pick from a list of arbitrary blocks, and stack and nest them in any order they like, it's going to be a significant challenge for the system to get the right margins and padding for every block in every context.

In particular, if there are examples like "if a video block follows a hero image, then the margin-bottom on the hero image should be X, but if a govspeak block follows a hero image then the margin-bottom on the hero image should be Y" then we're going to get into a really mucky place.

If it's possible for every block to have the same margins / padding as every other kind of block (at least at the top level), and for them to stack nicely in a way that doesn't ruin the designs, I think that's the ideal outcome.

If on the other hand, blocks need different margins in different contexts, I think it's probable more pragmatic to allow the user to provide their own margins (as we do here), perhaps while holding our noses a bit and telling ourselves "it's okay, someone from GDS will always review this content before it goes live".

Reply:
Allowing content authors control over spacing could introduce inconsistencies as well as allowing the possibility that they could overlook problems, particularly if margins are different at different breakpoints. I'd vote for the consistent spacing approach.

Reply:
Also, if we use a responsive margin in the selector above, then we can tweak the spacing for mobile/tablet/desktop while still maintaining consistency.

Reply:
The exception would be `govspeak` content/headings, which probably would not want the same amount of consistent bottom margin as typically these are related/connect to the following component. So we might need a general rule:

_All components/blocks get `margin-bottom: XX` when used as a direct child of the main content container, **except** govspeak components that are just text and relate to the following component. These should get `margin-bottom: YY`._